pkg/noobaaaccount/noobaaaccount.go

```diff
@@ -707,6 +769,116 @@ func GenerateNonCrdAccountKeys(name string) error {
+	secret := util.KubeObject(bundle.File_deploy_internal_secret_empty_yaml).(*corev1.Secret)
+	secret.Namespace = options.Namespace
+	// Handling a special case when the account is "admin@noobaa.io" we don't have CRD but have a secret
+	if name == "admin@noobaa.io" {
+		secret.Name = "noobaa-admin"
+	} else {
+		secret.Name = fmt.Sprintf("noobaa-account-%s", name)
+	}
+	if !util.KubeCheckQuiet(secret) {
+		log.Fatalf(`❌  Could not find secret: %s, will not update keys.`, secret.Name)
+	}
+
+	err = sysClient.NBClient.UpdateAccountKeysAPI(nb.UpdateAccountKeysParams{
+		Email:      name,
+		AccessKeys: accessKeys,
+	})
+	if err != nil {
+		return err
+	}
+
+	// UpdateAccountKeysAPI have no reply so we need to read the account in order to get the new credentials
+	accountInfo, err := sysClient.NBClient.ReadAccountAPI(nb.ReadAccountParams{
+		Email: name,
+	})
+	if err != nil {
+		return err
+	}
+
+	accessKeys = accountInfo.AccessKeys[0]
+
+	secret.StringData = map[string]string{}
+	secret.StringData["AWS_ACCESS_KEY_ID"] = accessKeys.AccessKey
+	secret.StringData["AWS_SECRET_ACCESS_KEY"] = accessKeys.SecretKey
+
+	// If we will not be able to update the secret we will print the credentials as they already been changed by the RPC
+	if !util.KubeUpdate(secret) {
+		log.Printf(`❌  Please verify the updated credentials for account %s:`, name)
+		fmt.Printf("\nAWS_ACCESS_KEY_ID     : %s\n", accessKeys.AccessKey)
+		fmt.Printf("AWS_SECRET_ACCESS_KEY : %s\n\n", accessKeys.SecretKey)
+		log.Fatalf(`❌  Failed to update the secret %s with the new accessKeys`, secret.Name)
+	}
+
+	log.Printf("✅ Successfully updated s3 credentials for the account %q", name)
+	return nil
+}
+
+// UpdateNonCrdAccountKeys update noobaa account (none CRD based) S3 keys
+func UpdateNonCrdAccountKeys(name string, accessKeys nb.S3AccessKeys) error {
+	log := util.Logger()
+
+	sysClient, err := system.Connect(true)
+	if err != nil {
+		return err
+	}
+
+	err = sysClient.NBClient.UpdateAccountKeysAPI(nb.UpdateAccountKeysParams{
+		Email:      name,
+		AccessKeys: accessKeys,
+	})
+	if err != nil {
+		if nbErr, ok := err.(*nb.RPCError); ok && nbErr.RPCCode == "NO_SUCH_ACCOUNT" {
+			log.Fatalf(`❌  Could not find the account: %s, will not update keys.`, name)
+		}
+		return err
+	}
+
+	// UpdateAccountKeysAPI have no reply so we need to read the account in order to get the updated credentials
+	accountInfo, err := sysClient.NBClient.ReadAccountAPI(nb.ReadAccountParams{
+		Email: name,
+	})
+	if err != nil {
+		log.Fatalf(`❌  Could not read account: %s, keys were allready updated, please read the account to get the keys`, name)
+	}
+
+	accessKeys = accountInfo.AccessKeys[0]
+
+	log.Printf("✅ Successfully update s3 credentials for the account %q", name)
+	fmt.Printf("\nAWS_ACCESS_KEY_ID     : %s\n", accessKeys.AccessKey)
+	fmt.Printf("AWS_SECRET_ACCESS_KEY : %s\n\n", accessKeys.SecretKey)
+
+	return nil
+}
+
+// ValidateAccessKeys checks for complexity of credentials(access_key and secret_key)
```

Comment:
In the core repo the term "complexity" was removed.

Reply:
I will remove it from operator. Thanks